src/Agent.Worker/Handlers/NodeHandler.cs

```diff
@@ -220,6 +221,15 @@ public async Task RunAsync()
             var sigtermTimeout = TimeSpan.FromMilliseconds(AgentKnobs.ProccessSigtermTimeout.GetValue(ExecutionContext).AsInt());
             var useGracefulShutdown = AgentKnobs.UseGracefulProcessShutdown.GetValue(ExecutionContext).AsBoolean();
 
+            var debugTask = AgentKnobs.DebugTask.GetValue(ExecutionContext).AsString();
+            if (!string.IsNullOrEmpty(debugTask))
```

Comment:
It would be cool to detect if the agent is running in debug mode and enable this only then, however I am not aware of such global setting today.